tests/rptest/clients/types.py

```diff
@@ -210,3 +210,10 @@ def _random_cleanup_policy() -> str:
                 TopicSpec.CLEANUP_DELETE,
             ]
         )
+
+
+# A topic represented as either a spec or a simple name.
+# This is useful in APIs that accept either and often just
+# use __str__ (or use it in an f-string, etc) which will
+# work for both.
+AnyTopic = str | TopicSpec
```

Comment:
This is a bit ugly no? When importin confluent-kafka I just changed any TopicSpec calls to use the `.name` attribute. Would that be too much work to change?

Reply:
> I don't think it's ugly if you want to have APIs which accept both str and TopicSpec and just work, then this is the pythonic way to do it (and that kind of thing is pretty pythonic itself).

I mean sure but like you say I feel like this is mostly accidental and only allowing `str` seems sensible to me. Given `__str__` for TopicSpec could really change at any time. If the method does TopicSpec.name then it's fine but relying on __str__ to effectively be an alias for .name seems ugly to me.

Reply:
And in the confluent-kafka case we passed TopicSpec to confluent-kafke which knows nothing about what TopicSpec is.